google/cloud/status.h

```diff
@@ -36,25 +36,162 @@ enum class StatusCode {
+  /// raised by APIs that do not return enough error information may be
+  /// converted to this error.
   kUnknown = 2,
+
+  /// `kInvalidArgument` (gRPC code `INVALID_ARGUMENT`) indicates the caller
+  /// specified an invalid argument, such as a malformed filename.
+  ///
+  /// Note that use of such errors should be narrowly limited to indicate the
+  /// invalid nature of the arguments themselves. Errors with validly formed
+  /// arguments that may cause errors with the state of the receiving system
+  /// should be denoted with `kFailedPrecondition` instead.
   kInvalidArgument = 3,
+
+  /// `kDeadlineExceeded` (gRPC code `DEADLINE_EXCEEDED`) indicates a deadline
+  /// expired before the operation could complete.
+  ///
+  /// For operations that may change state within a system, this error may be
+  /// returned even if the operation has completed successfully. For example, a
+  /// successful response from a server could have been delayed long enough for
+  /// the deadline to expire.
   kDeadlineExceeded = 4,
+
+  /// `kNotFound` (gRPC code `NOT_FOUND`) indicates some requested entity (such
+  /// as
+  /// a file or directory) was not found.
+  ///
+  /// `kNotFound` is useful if a request should be denied for an entire class of
+  /// users, such as during a gradual feature rollout or undocumented allow
+  /// list.
+  /// If a request should be denied for specific sets of users, such as through
+  /// user-based access control, use `kPermissionDenied` instead.
   kNotFound = 5,
+
+  /// `kAlreadyExists (gRPC code `ALREADY_EXISTS`) indicates that the entity a
+  /// caller attempted to create (such as a file or directory) is already
+  /// present.
   kAlreadyExists = 6,
+
+  /// `kPermissionDenied` (gRPC code `PERMISSION_DENIED`) indicates that the
+  /// caller does not have permission to execute the specified operation.
+  ///
+  /// Note that this error is different than an error due to an
+  /// *un*authenticated caller. This error code does not imply the request is
+  /// valid or the requested entity exists or satisfies any other
+  /// pre-conditions.
+  ///
+  /// `kPermissionDenied` must not be used for rejections caused by exhausting
+  /// some resource. Instead, use `kResourceExhausted` for those errors.
+  /// `kPermissionDenied` must not be used if the caller cannot be identified.
+  /// Instead, use `kUnauthenticated` for those errors.
   kPermissionDenied = 7,
-  kUnauthenticated = 16,
+
+  /// `kResourceExhausted` (gRPC code `RESOURCE_EXHAUSTED`) indicates some
+  /// resource has been exhausted.
+  ///
+  /// Examples include a per-user quota, or the entire file system being out of
+  /// space.
   kResourceExhausted = 8,
+
+  /// `kFailedPrecondition` (gRPC code `FAILED_PRECONDITION`) indicates that the
+  /// operation was rejected because the system is not in a state required for
+  /// the operation's execution.
+  ///
+  /// For example, a directory to be deleted may be non-empty, an "rmdir"
```

Comment:
Nit: We say `a "rmdir"` below, which sounds better to my "rrrrrrmdir" pronounciation.

Reply:
Fixed